### PR TITLE
break out read log part of measureme to the tools_lib crate

### PR DESCRIPTION
**Pull request by andjo403**

### CODE REVIEW

**Comment by andjo403** [comment]
not super happy about the solution with having the testing_common.rs in the tools_lib as it feels like it is only for tests in measureme. one possibility is to move out the bench and test folder from measureme but it feels like they belong in measureme

**Comment by wesleywiser** [comment]
@michaelwoerister would you like to review as well? This moves the code which reads profiling data into the `tools_lib` crate which is nice because it allows us to add helpful dependencies such as `serde` without increasing the size of the `measureme` crate.

File: measureme/Cargo.toml [hunk 1]
```diff
@@ -18,3 +18,6 @@ rustc-hash = "1.0.1"
 
 [target.'cfg(not(target_arch="wasm32"))'.dependencies]
 memmap = "0.6.0"
+
+[dev-dependencies]
+tools_lib = { path = "../tools_lib" }
```
Comment (michaelwoerister) [inline measureme/Cargo.toml]: I'd like to get rid of this (pseudo)circular dependency. I think it's fine to move the end-to-end serialization tests into `tools_lib`.

File: tools_lib/src/lib.rs [hunk 2]
```diff
@@ -1 +1,21 @@
...
+//! # Reading event trace files
+//!
+//! The main entry point for reading trace files is the [`ProfilingData`] struct.
+//!
+//! To create a [`ProfilingData`], call the [`ProfilingData::new()`] function and provide a `Path` with the directory and file name for the trace files.
+//!
+//! To retrieve an `Iterator` of all of the events in the file, call the [`ProfilingData::iter()`] method.
+//!
+//! [`ProfilingData`]: struct.ProfilingData.html
+//! [`ProfilingData::iter()`]: struct.ProfilingData.html#method.iter
```
Comment (michaelwoerister) [inline tools_lib/src/lib.rs]: This doc string could use some line breaks.

**Comment by andjo403** [comment]
fixed the comments

**Comment by michaelwoerister** [comment]
Thanks, @andjo403!

**Pull request merged**